sdl2promela/sdl/model.py

```diff
@@ -399,15 +550,77 @@ def convert(source: ogAST.TaskAssign):
         )
     task = AssignmentTask()
     task.assignment = convert(source.elems[0])
+    task.type = source.elems[0].left.exprType
     return task
 
 
 @dispatch(ogAST.PrimVariable)
 def convert(source: ogAST.PrimVariable):
-    variableReference = VariableReference(source.inputString)
+    variableReference = VariableReference(source.value[0])
+
     return variableReference
 
 
+@dispatch(ogAST.PrimSequence)
+def convert(source: ogAST.PrimSequence):
+    elements: Dict[str, Expression] = {}
+    for elem, value in source.value.items():
+        elements[elem] = convert(value)
+
+    return Sequence(elements, source.exprType)
+
+
+@dispatch(ogAST.PrimSequenceOf)
+def convert(source: ogAST.PrimSequenceOf):
+    elements: List[Expression] = []
+    for elem in source.value:
+        elements.append(convert(elem))
+
+    return SequenceOf(elements, source.exprType)
+
+
+@dispatch(ogAST.PrimEmptyString)
+def convert(source: ogAST.PrimEmptyString):
+    return EmptyStringValue()
+
+
+@dispatch(ogAST.PrimStringLiteral)
+def convert(source: ogAST.PrimStringLiteral):
+    return StringValue(source.value[1:-1])
+
+
+@dispatch(ogAST.PrimOctetStringLiteral)
+def convert(source: ogAST.PrimOctetStringLiteral):
+    characters = list(source.value[1:-1])
+    length = len(characters)
+    elements = []
+
+    for index in range(0, length, 2):
+        byte_value = int("".join(characters[index : index + 2]), 16)
```

Comment:
bytes.fromhex(hex_string)? If not, maybe extract....

Reply:
Done.